### DISCUSSION TITLE
More wildcards

### DISCUSSION THREAD

**Discussion by MoonRide303** [lllyasviel/Fooocus#741] [Ideas]
I would like to discuss enhancing list of build-in wildcards here.

Requirements:
- it should be useful in context of Fooocus (proposed concepts should be known by CLIP model, so the conditioning could work),
- it has to be high quality (reliable source and good coverage of given subject / concept),
- it should be explainable and reproducible (another person using same logic could end up with same list),
- it should stay SFW (so people could use Fooocus in schoold or at work without getting into troubles).

As an example I will briefly describe existing wildcards that are already available in current version of Fooocus (v2.1.709):
1. color.txt - names of commonly used web colors, source: [CSS 2.2 specification](https://www.w3.org/TR/CSS22/syndata.html#color-units).
2. artist.txt - list of artist names recognized by CLIP model, source: [Image Synthesis Style Studies Database (The List)](https://docs.google.com/spreadsheets/d/14xTqtuV3BuKDNhLotB_d1aFlBGnDJOY0BRXJ8-86GpA/edit?usp=sharing) (filtered out by CLIP).

If you would like to have more wildcards, please throw it into this discussion - just make sure requirements are met, and provide description similar to those 2 examples above.

**Comment by mmmz28** [comment]
![image](https://github.com/lllyasviel/Fooocus/assets/58453940/cd1765f8-fc88-4f56-a2e3-07e2a0694541)
Reproducible is important! When I use "__color__ cat in a __color__ hat with a __color__ eyes,by __artist__" to get a style image that I like, I don't know which artist it is. The log.html has no record and is still "by __artist__", not the specific artist
![截取图片_20231020151543](https://github.com/lllyasviel/Fooocus/assets/58453940/1612abc4-6a4a-4095-bd99-4efd922c8fd3)

**Reply by MoonRide303** [reply]
@mmmz28 I've added information about wildcards usage in v2.1.710 (https://github.com/lllyasviel/Fooocus/pull/742) - if wildcards were used it will be printed in console log, like that:
```
[Fooocus Wildcards] input: __color__ cat in a __color__ hat with a __color__ eyes, by __artist__ and __artist__
[Fooocus Wildcards] output: red cat in a fuchsia hat with a lime eyes, by Adrian Tomine and Rinko Kawauchi
```

**Reply by MoonRide303** [reply]
And also in the Private Log, added in v2.1.711 (https://github.com/lllyasviel/Fooocus/pull/743):
![image](https://github.com/lllyasviel/Fooocus/assets/130458190/1bd81ae9-1d8f-4211-9478-8ce18a901f72)

**Reply by MoonRide303** [reply]
And simplified a bit in v2.1.712 (most users won't be using wildcards and double prompt information would be confusing - so I removed raw prompt from the log).

**Reply by rubenet12** [reply]
Hi @MoonRide303 
would it be possible to get the prompt used in the log when using wildcards?
I am using wildcards to mix animals and loved some of the results... but I can´t try to reproduce or work on that image without knowing which value was used.
I checked the log file as a text file and can´t find a the prompt version showing the actual values, just in case it was hidden.
Is there any chance to get a workaround or to include both raw and prompt in the log?
Thanks,
Ruben

**Comment by agarzon** [comment]
Just added a PR https://github.com/lllyasviel/Fooocus/pull/748

I think it fits the requirements.

**Reply by MoonRide303** [reply]
@agarzon While ChatGPT is not really reliable source of information (it sometimes lies and/or don't provide complete information) it looked good enough to be merged - we can update it in future, if needed.
